beta-src/src/components/WDPopover.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import { useRef } from "react";
+import { Box, Popover } from "@mui/material";
+
+interface WDPopoverProps {
+  children: React.ReactNode;
+  isOpen: boolean;
+  /**
+   *A callback that sets isOpen to false
```

Comment:
it may not do that. this callback could do anything. we need to just say:

a callback to be used when the popover closes

or something to that effect.